chain/chain/src/chain.rs

```diff
@@ -782,16 +792,27 @@ impl Chain {
         } else {
             debug!(target: "chain", "Downloading state for {:?}, I'm {:?}", shards_to_state_sync, me);
 
-            let state_sync_info = StateSyncInfo {
-                epoch_tail_hash: *block.header().hash(),
+            let protocol_version = self
+                .epoch_manager
+                .get_epoch_protocol_version(epoch_first_block.header().epoch_id())?;
+            let sync_hash = if ProtocolFeature::StateSyncHashUpdate.enabled(protocol_version) {
+                None
+            } else {
+                Some(*epoch_first_block.header().hash())
+            };
```

Comment:
Ah no here actually it's not possible because we're inside `preprocess_block()`, so `epoch_first_block` has not yet been processed and saved as HEAD. So then if you try calling `get_sync_hash(epoch_first_block.header().hash())`, you'll get a `DBNotFoundErr` when looking up that block's hash.